code/processes/wdb.q

```diff
@@ -296,8 +308,10 @@ merge:{[dir;pt;tableinfo;mergelimits]
 endofdaymerge:{[dir;pt;tablist;mergelimits]		
 	/- merge data from partitons
 	$[(0 < count .z.pd[]) and ((system "s")<0);
-		[.lg.o[`merge;"merging on slave"];
+		[.lg.o[`merge;"merging on slave"];		
+		setcompression[compression];
 		merge[dir;pt;;mergelimits] peach flip (key tablist;value tablist);];	
+		removecompression[compression];
 		[.lg.o[`merge;"merging on master"];
 		merge[dir;pt;;mergelimits] each flip (key tablist;value tablist)]];
```

Comment:
I think you need to wrap this block in set/remove statements?  This is the bit that does the merging on the master process